apidoc/lib/typescript_generator.js

```diff
@@ -0,0 +1,542 @@
+'use strict';
+
+/*
+ * Map of invalid types and their replacement
+ */
+const invalidTypeMap = {
+	'2DMatrix': 'Matrix2D',
+	'3DMatrix': 'Matrix3D',
+	'Dictionary': 'any'
```

Comment:
in our parlance a "Dictionary" is an Object, I believe.

Reply:
In TypeScript the `object` type only refers to the Object prototype and its properties and methods. Our `Dictionary` refers to any custom object or a specific set of properties/methods of another type, for which i haven't found a proper way of representing that in TypeScript yet. So i just replace it with `any` for now.